app/models/task.py

```diff
@@ -2,4 +2,27 @@
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    is_completed = db.Column(db.Boolean, default=False)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    def to_dict(self):
+        return {
+            "id": self.task_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.is_completed
+        }
+    
+    def to_dict_in_goal(self):
+        return {
+            "id": self.task_id,
+            "goal_id": self.goal_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.is_completed
+        }
```

Comment:
Nice helper functions! We can condense these two functions into one by building an initial dictionary and then checking if a goal_id exists for that particular task.

```python
  def to_dict(self):
      task = {
            "id": self.task_id,
            "title": self.title,
            "description": self.description,
            "is_complete": self.is_completed
        }

      if self.goal_id:
             task['goal_id'] = self.goal_id

      return task

```